### PR TITLE
New upgrade script for Python.

### PR DESCRIPTION
**Pull request by ludeeus**
## Description:
This script will upgrade Python to the latest stable version.
It will also create a new virtual environment to be used for Home Assistant.
If it fails, it will revert back to the original virtual environment.

## Checklist:
- [X] The code change is tested and works locally.
  - [X] Hassbian 1.4.0 "Stretch" (From Python 3.5.3)
  - [X] Hassbian 1.23 "Jessie" (From Python 3.4.2)

Both tested with current dev branch of `hassbian-scripts`
- [X] Script has validation check of the job.
  
### If pertinent:
  - [X] Created/Updated documentation at `/docs`

### CODE REVIEW

**Comment by ludeeus** [comment]
I'm unsure about this, it can help many users with the upgrade, but it's also a big change.
Maybe add this to the script?
Requiring users to take another step.
```bash
if [ "$FORCE" == "" ]; then
    echo "This script will change a lot on your pi."
    echo "DO NOT run this if you absolutely have to."
    echo "You can force run the upgrade script like this:"
    echo "sudo hassbian-config upgrade python --force"
   return 0
  fi
```

**Comment by Landrash** [comment]
I'm unsure about this script. What's your reason for wanting it in?

File: docs/python.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Description
+This script will upgrade Python to the latest stable version.  
+It will also create a new virtual environment to be used for Home Assistant.  
+_This upgrade takes loong time to finish._
```
Comment (Landrash) [inline docs/python.md]: Spelling of long.

File: docs/python.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+## Description
+This script will upgrade Python to the latest stable version.  
+It will also create a new virtual environment to be used for Home Assistant.  
+_This upgrade takes loong time to finish._  
+_This prosess takes about a hour on an RPi 3_
```
Comment (Landrash) [inline docs/python.md]: Change RPI 3 to Rapsberry Pi 3.

File: docs/python.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+## Description
```
Comment (Landrash) [inline docs/python.md:1]: Add information that this is not recommended.

**Review by Landrash — CHANGES_REQUESTED** [review]
Some minor cosmetic tweaks and discussion about why this should be added should probably be had.

**Comment by ludeeus** [comment]
The main reason is to supply an easy way of upgrading the venv that Home Assistant runs in to a least be supported on jessie (older Hassbian devices), but this will install the latest stable version of Python to comply with future compability issues, and give the benefits that Python 3.6.x have over 3.5.x.

@Landrash Since both you and I is unsure about including this, I suggest we closed it for now and revisit it later if we want :)

**Comment by Landrash** [comment]
I'm actually for this but it would be nice if you make this not run without the --force command and make it print out a warning if you try to install it anyway.

File: docs/python.md (hunk 3 above)
Comment (ludeeus) [reply docs/python.md:1]: Better now? :)

**Review by Landrash — APPROVED** [review]

**Pull request merged**